server/src/services/animal-location.service.ts

```diff
@@ -89,19 +89,16 @@ export class AnimalLocationService {
         });
     }
 
-    getAllAnimalsLocationTableData(): any {
-
-        const con =  this.databaseService.getConnection();
-        return con.then(async (data)=>{
-           return await data.getRepository(AnimalLocation).find(); 
-        })  
+    async getAllAnimalsLocationTableData(): Promise<JSON> {
+        const con = await this.databaseService.getConnection();
+        let animaldata =  JSON.parse(JSON.stringify( await con.getRepository(AnimalLocation).find()));
+        return animaldata;
     }
 
-    getIndividualAnimalLocationTableData(animalID): any {
-
-        const con =  this.databaseService.getConnection();
-        return con.then(async (data)=>{
-           return await con.getRepository(AnimalLocation).find({ animalId : animalID }); 
+    async getIndividualAnimalLocationTableData(animalID): Promise<JSON> {
+        const con = await this.databaseService.getConnection();
+        let animaldata =  JSON.parse(JSON.stringify( await con.getRepository(AnimalLocation).find({ animalId : animalID })));
```

Comment:
You can just do ```return JSON.parse(JSON.stringify(...))``` instead of assigning to a variable and then immediately returning that variable. Also remove unneccessary spaces

Reply:
The change you proposed makes more sense, will do this